libs/agency/feature/dashboard/src/lib/components/dashboard/dashboard-service.ts

```diff
@@ -31,7 +31,12 @@ export class DashboardService {
     return combineLatest([
       this.getDashboards$(),
       this._activatedRoute.queryParamMap.pipe(
-        map(params => params.get('transactionSet') ?? ''),
+        map(params => {
+          const transactionSet = params.get('transactionSet') ?? '';
```

Comment:
A couple thing here, I think we can simplify this and load and set the item all in the same place here
- We should use a const for the name ex. `const TRANSACTION_SET_STORAGE_KEY = 'transactionSet';`

If you do these changes, you should need any of the other changes


```suggestion
          const transactionSet = params.get('transactionSet') ?? localStorage.getItem('transactionSet') ?? '';
```